lib/semian/atomic_enum.rb

```diff
@@ -0,0 +1,38 @@
+require 'forwardable'
+
+module Semian
+  class AtomicEnum #:nodoc:
+    extend Forwardable
+
+    def_delegators :@integer, :destroy
+
+    def initialize(symbol_list)
+      @integer = Semian::AtomicInteger.new
+      initialize_lookup(symbol_list)
+    end
+
+    def increment(val = 1)
+      @integer.value = (@integer.value + val) % @sym_to_num.size
```

Comment:
Here's what it should look like: 
https://github.com/Shopify/semian/blob/circuit-breaker-per-host/lib/semian/atomic_integer.rb

Notice the `surround_with_mutex`, ... but I threw that in `SharedMemoryObject`